plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2375,6 +2392,46 @@ private static DeltaLakeColumnHandle createProjectedColumnHandle(DeltaLakeColumn
                 Optional.of(projectionInfo));
     }
 
+    /**
+     * Returns the assignment key corresponding to the column represented by {@param projectedColumn} in the {@param assignments}, if one exists.
+     * The variable in the {@param projectedColumn} can itself be a representation of another projected column. For example,
+     * say a projected column representation has variable "x" and a dereferenceIndices=[0]. "x" can in-turn map to a projected
+     * column handle with base="a" and [1, 2] as dereference indices. Then the method searches for a column handle in
+     * {@param assignments} with base="a" and dereferenceIndices=[1, 2, 0].
+     */
+    private static Optional<String> find(Map<String, ColumnHandle> assignments, ProjectedColumnRepresentation projectedColumn)
+    {
+        DeltaLakeColumnHandle variableColumn = (DeltaLakeColumnHandle) assignments.get(projectedColumn.getVariable().getName());
+
+        if (variableColumn == null) {
+            return Optional.empty();
```

Comment:
is this impossible case?

Reply:
yes, It is impossible case. Method is Copy of HiveApplyProjectionUtil#find. Replaced with `requireNonNull(variableColumn, "variableColumn is null");`